### PR TITLE
Manga page display 

### PR DESCRIPTION
**Pull request by fairnightzz**
pls do other pr first

### CODE REVIEW

File: src/pages/Reader.tsx [hunk 1]
```diff
@@ -29,47 +32,78 @@ class Reader extends React.Component<any, State> { // fix typing up
...
+
+    // this.setState(curState => {
+    //   return {
+    //     ...curState,
+    //     images: this.state.images.splice(pageNumber, 1, rawImg)
+    //   }
+    // });
+
+    if (pageNumber === -1) {
+      console.log(this.state.chapters, 'chapters');
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx:91]: try to use console.debug whenever you can

File: src/pages/Reader.tsx [hunk 2]
```diff
@@ -29,47 +32,78 @@ class Reader extends React.Component<any, State> { // fix typing up
...
+      console.log(this.state.pages, 'pages');
+      // load everything
+      const images = [];
+      for (let i = 0; i < this.state.pages.length; i += 1) {
+        images.push(`https://files.karatsubascans.com/${this.state.mangafile}/jpg/`+
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx:96]: maybe move the `https://file/karatsubascans.com` part to its own API_URL config variable

File: src/pages/Reader.tsx [hunk 3]
```diff
@@ -29,47 +32,78 @@ class Reader extends React.Component<any, State> { // fix typing up
...
+          `${this.state.chapters[this.state.chapter].name}/`+
+          `${this.state.pages[i].name}`);
       }
-    });
+
+      this.setState({
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx:101]: isn't this overriding the state?

File: src/pages/Reader.tsx [hunk 4]
```diff
@@ -29,47 +32,78 @@ class Reader extends React.Component<any, State> { // fix typing up
...
+    //     images: this.state.images.splice(pageNumber, 1, rawImg)
+    //   }
+    // });
+
+    if (pageNumber === -1) {
+      console.log(this.state.chapters, 'chapters');
+      console.log(this.state.pages, 'pages');
+      // load everything
+      const images = [];
+      for (let i = 0; i < this.state.pages.length; i += 1) {
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx:95]: nitpicky but just use `++i` instead of `i += 1`

File: src/utils/requests.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import axios from 'axios';
+import { Chapter, Page } from 'types/reader.types';
+
+export const getPages = async (mangafile: string, chapter: string): Promise<Page[]> => {
+
+    const url = `https://files.karatsubascans.com/${mangafile}/jpg/${chapter}`;
```
Comment (MrPicklePinosaur) [inline src/utils/requests.tsx:6]: ok at this point definitely make a variable for API_URL

**Review by MrPicklePinosaur — CHANGES_REQUESTED** [review]
LGTM, except the comments

**Comment by MrPicklePinosaur** [comment]
also why did so much of package-lock.json change, try to use `npm ci` instead of `npm i` if you can to avoid changing package versions

File: src/pages/Reader.tsx (hunk 1 above)
Comment (fairnightzz) [reply src/pages/Reader.tsx:91]: bruh

File: src/pages/Reader.tsx (hunk 3 above)
Comment (fairnightzz) [reply src/pages/Reader.tsx:101]: nop it is not, idk why u were doing some of the arrow stuff

File: src/pages/Reader.tsx (hunk 4 above)
Comment (fairnightzz) [reply src/pages/Reader.tsx:95]: wha airbnb doesnt like `++` in general

Comment (fairnightzz) [reply src/pages/Reader.tsx:95]: https://airbnb.io/javascript/#variables--unary-increment-decrement

**Pull request merged**